code/game/gamemodes/miniantags/morph/spells/morph_spell.dm

```diff
@@ -0,0 +1,36 @@
+/obj/effect/proc_holder/spell/morph
+	action_background_icon_state = "bg_morph"
+	clothes_req = FALSE
+	/// How much food it costs the morph to use this
+	var/hunger_cost = 0
+
+/obj/effect/proc_holder/spell/morph/Initialize(mapload)
+	. = ..()
+	if(hunger_cost)
+		name = "[name] ([hunger_cost])"
+
+/obj/effect/proc_holder/spell/morph/cast_check(charge_check = TRUE, start_recharge = TRUE, mob/living/simple_animal/hostile/morph/user = usr)
+	if(!istype(user))
+		to_chat(user, "<span class='warning'>You should not be able to use this abilty! Report this as a bug on github please.</span>")
+		stack_trace()
+		log_debug("[user] has the spell [src] while he is not a morph")
+		return FALSE
+	if(user.gathered_food < hunger_cost)
+		to_chat(user, "<span class='warning'>You require at least [hunger_cost] stored food to use this ability!</span>")
+		return FALSE
+	return ..()
+
+/obj/effect/proc_holder/spell/morph/can_cast(mob/living/simple_animal/hostile/morph/user, charge_check, show_message)
+	if(!istype(user) || user.gathered_food < hunger_cost)
+		return FALSE
+	return ..()
+
+/obj/effect/proc_holder/spell/morph/before_cast(list/targets, mob/living/simple_animal/hostile/morph/user)
+	user.use_food(hunger_cost)
+	if(hunger_cost)
+		to_chat(user, "<span class='boldnotice'>You have [user.gathered_food] left to use.</span>")
```

Comment:
The way this is set up makes abilities which cost food give confusing feedback in chat: 

![image](https://user-images.githubusercontent.com/42044220/103852766-30ca7300-5072-11eb-9edc-3ee107148cee.png)

`user.use_food(hunger_cost)` should probably go further down in the chain of spell procs or something.

Reply:
Aye I saw that as well while testing. Issue with fixing this is that I have to refactor the spell code even more :(
The costs are currently deducted before a cast. So if the user cancels the cast midway they will get a refund then the message will make more sense. Currently the `after_cast` proc is also always called no matter what `cast` returns so putting it there will lead to even worse messages. You have 30 left. You have 20 left (since it gets reversed before the `after_cast` gets called.

I think I'm gonna leave this one in till that refactor

Reply:
Ah alright, I don't blame you, spell code is confusing/annoying to work with. Refactor is out of scope.